Validation/TrackerRecHits/src/SiStripRecHitsValid.cc

```diff
@@ -594,26 +584,31 @@ void SiStripRecHitsValid::createMEs(DQMStore::IBooker & ibooker,const edm::Event
     std::string label = hidmanager.getSubdetid(detid,tTopo,true);
     // std::cout << "label " << label << endl;
       
-    std::map<std::string, LayerMEs>::iterator iLayerME  = LayerMEsMap.find(label);
-    if(iLayerME==LayerMEsMap.end()) {
+    if(LayerMEsMap.find(label)==LayerMEsMap.end()) {
 	
       // get detids for the layer
       // Keep in mind that when we are on the TID or TEC we deal with rings not wheel 
       int32_t lnumber = det_layer_pair.second;
-      std::vector<uint32_t> layerDetIds;        
-      if (det_layer_pair.first == "TIB") {
-	substructure.getTIBDetectors(activeDets,layerDetIds,lnumber,0,0,0);
-      } else if (det_layer_pair.first == "TOB") {
+      std::string lname = det_layer_pair.first; 
+      std::vector<uint32_t> layerDetIds;
+      std::string tec = "TEC", tid = "TID", tob = "TOB", tib = "TIB";        
```

Comment:
no, this I asked for!
so that the comparison is string against string, not string against const char *